### PR TITLE
(refactor) - gold Hooks (-11B) and compat (-18B)

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Thought these packages deserved some attention aswell :D

### CODE REVIEW

File: compat/src/index.js [hunk 1]
```diff
@@ -15,8 +15,7 @@ options.event = e => {
 	/* istanbul ignore next */
 	if (oldEventHook) e = oldEventHook(e);
 	e.persist = () => {};
-	e.nativeEvent = e;
-	return e;
+	return e.nativeEvent = e;
```
Comment (marvinhagemeister) [inline compat/src/index.js]: neat :+1:

File: hooks/src/index.js [hunk 2]
```diff
@@ -44,9 +43,9 @@ options.unmount = vnode => {
 	if (!c) return;
 
 	const hooks = c.__hooks;
-	if (!hooks) return;
-
-	hooks._list.forEach(hook => hook._cleanup && hook._cleanup());
+	if (hooks) {
+		hooks._list.some(hook => hook._cleanup && hook._cleanup());
```
Comment (ForsakenHarmony) [inline hooks/src/index.js]: what if this returns a truthy value?

Comment (JoviDeCroock) [reply hooks/src/index.js]: You are 100% right, I'll fix this tonight

Comment (marvinhagemeister) [reply hooks/src/index.js]: @ForsakenHarmony good catch, missed that in my review :+1:

**Pull request merged**